flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/ManuallyTriggeredScheduledExecutorService.java

```diff
@@ -318,11 +319,22 @@ public void triggerNonPeriodicScheduledTasks(Class<?> taskClazz) {
     public void triggerPeriodicScheduledTasks() {
         for (ScheduledTask<?> scheduledTask : periodicScheduledTasks) {
             if (!scheduledTask.isCancelled()) {
-                scheduledTask.execute();
+                executeScheduledTask(scheduledTask);
             }
         }
     }
 
+    private static void executeScheduledTask(ScheduledTask<?> scheduledTask) {
+        scheduledTask.execute();
+        try {
+            // try to retrieve result of scheduled task to avoid swallowing any exceptions that
+            // occurred
+            scheduledTask.get();
```

Comment:
This blocking call breaks a lot of stuff. I don't think we can make this change; too many tests rely on throwing something in the executor, and later triggering the completion once some condition is fulfilled.

Maybe handle exceptions as fatal errors tho. Maybe that breaks fewer things.

Reply:
This doesn't work for periodic tasks since the result future in the ScheduledTask never gets completed.

I'm not sure if this change is correct in the first place. Every non-periodic task already throws exceptions when trigger is called.

Reply:
True, I missed that we're not retrieving the result in case of periodic tasks in [ScheduledTask#execute](https://github.com/apache/flink/blob/5405239dec0884dff746129c73955c90f455c465/flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/ScheduledTask.java#L76).

I changed the behavior: Instead of collecting the error in the future, we're going to throw the error rightaway:
> ScheduledTask only serves as a container for a Callable. The error handling should be done by an Executor (e.g. the main thread). Therefore, explicitly handling errors inside #execute is out of scope for ScheduledTask.